### PR TITLE
Token decoding

### PR DESCRIPTION
**Pull request by timkurvers**
Certain identity providers (looking at you Azure AD) may have multiple public keys in use simultaneously, which is why `Token#decode!` iterates over these public keys, attempting to decode and validate the given token. If no public key applied, an error will be raised.

Passing `:skip_verification` is supported by the internal OpenID connect client to skip validating signatures altogether.

Note that this includes PR #24 as it relies on it.

### CODE REVIEW

**Comment by timkurvers** [comment]
Don't merge this yet, going to fix that begin/rescue-ception.

**Comment by timkurvers** [comment]
This is ready for merging now :smile:

File: lib/openid_token_proxy/token.rb [hunk 1]
```diff
@@ -11,5 +11,38 @@ def initialize(access_token, id_token = nil, refresh_token = nil)
     def to_s
       @access_token
     end
+
+    # Raised when a token was not provided
+    class TokenRequired < StandardError; end
```
Comment (toothfairy) [inline lib/openid_token_proxy/token.rb]: Shouldn't that exceptions extend new Error class?

Comment (timkurvers) [inline lib/openid_token_proxy/token.rb]: Yes, will update! :)

**Pull request merged**